lib/kubeclient/common.rb

```diff
@@ -202,6 +205,14 @@ def resource_name(entity_type)
         entity_type.pluralize.downcase
       end
 
+      def proxy_url_service(namespace, service, protocol)
```

Comment:
OK so you've just found helped me find a bug in OpenShift/Kubernetes :) Services with an unnamed single port (totally allowed) don't require a port name for proxying; if the port has a name, even if it's single port, then it's required for proxying.